### ISSUE TITLE
Assertion Fail in chainSeedsGlobally

### ISSUE THREAD

**Issue by tloka** [seqan/seqan#2340]
At ECCB workshop, we had the following assertion failing when implementing lagan and compiling in debug mode:
```
/media/lokat/DATA/seqan_tutorial/tutorial/seqan/include/seqan/seeds/seeds_global_chaining.h:281 Assertion failed : _checkScoreInvariant(intermediateSolutions) should be true but was 0

stack trace:
  0          [0x67d89b]  seqan::ClassTest::fail() + 0xe
  1          [0x69d50d]  void seqan::chainSeedsGlobally<seqan::String<seqan::Seed<seqan::Tag<seqan::Chained_>, seqan::DefaultSeedConfig>, seqan::Alloc<void> >, seqan::Seed<seqan::Tag<seqan::Chained_>, seqan::DefaultSeedConfig>, seqan::Tag<seqan::Unordered_> >(seqan::String<seqan::Seed<seqan::Tag<seqan::Chained_>, seqan::DefaultSeedConfig>, seqan::Alloc<void> >&, seqan::SeedSet<seqan::Seed<seqan::Tag<seqan::Chained_>, seqan::DefaultSeedConfig>, seqan::Tag<seqan::Unordered_> > const&, seqan::Tag<seqan::SparseChaining_> const&) + 0x96a
  2          [0x67c4c1]  main + 0x58c
  3    [0x7ff90e310830]  __libc_start_main + 0xf0
  4          [0x67a819]  _start + 0x29

Aborted (core dumped)
```

Corresponding code fragment:
```
using TSeed = Seed<ChainedSeed>;
using TSeedSet = SeedSet<TSeed>;
[...]
TSeedSet seedSet;
[...] // local chaining to seedSet
// global chaining
String<TSeed> global_chain;
chainSeedsGlobally( global_chain, seedSet, SparseChaining() );
```

**Comment by smehringer** [comment]
Hi @tloka,

thanks for reporting this. I think this is the same issue as #2318 and was fixed in PR #2330, which still needs to be merged though.
As soon as the fix is merged we will report back to you so you can see if it works fine then.

**Comment by smehringer** [comment]
Hi @tloka ,
we fixed several bugs in the chaining algorithm and everything should work as expected now. If you find some time you might want to try if your example works with the current develop branch.
